homeassistant/components/sensor/__init__.py

```diff
@@ -366,7 +366,7 @@ def _is_valid_suggested_unit(self, suggested_unit_of_measurement: str) -> bool:
         because a unit converter supports both.
         """
         # No need to check the unit converter if the units are the same
-        if self.native_unit_of_measurement == suggested_unit_of_measurement:
+        if self.__native_unit_of_measurement_compat == suggested_unit_of_measurement:
```

Comment:
@emontnemery 

I am not sure how to update the docstrings that you mentionned in your comment https://github.com/home-assistant/core/pull/151956#discussion_r2348070067